docs/decisions/0031-feature-branch-strategy.md

```diff
@@ -0,0 +1,93 @@
+- Good, because it promotes smaller, incremental Pull Requests (PRs), simplifying review processes.
+- Good, because it reduces the risk of major bugs being merged into the main branch.
+- Good, because it makes the process of integrating features into the main branch easier and faster.
+- Bad, potentially, if not managed properly, as it can lead to outdated branches if not regularly synchronized with the main branch.
+
+## Validation
+
+The effectiveness of the feature branch strategy will be validated through regular code reviews, successful CI/CD pipeline runs, and feedback from the development team.
+
+## Connector/Model Priorities
+
+Currently we are looking for community support on the following models
+
+The support on the below can be either achieved creating a practical example using one of the existing Connectors against one of this models or providing a new Connector that supports a platform that hosts one of the models below:
+
+| Model Name | Local Support | Platforms                            | Connectors                                    |
+| ---------- | ------------- | ------------------------------------ | --------------------------------------------- |
+| Gpt-4      | No            | OpenAI, Azure Open AI                | OpenAI                                        |
+| Phi-2      | Yes           | Azure OpenAI, Hugging Face, LMStudio | OpenAI, HuggingFace, LMStudio\*\*             |
+| Gemini     | No            | Google AI Platform                   | GoogleAI\*\*                                  |
+| Llama-2    | Yes           | LMStudio, HuggingFace, Ollama        | HuggingFace, OpenAI, LMStudio\*\*, Ollama\*\* |
+| Mistral    | Yes           | LMStudio, HuggingFace, Ollama        | HuggingFace, OpenAI, LMStudio\*\*, Ollama\*\* |
+| Claude     | No            | Anthropic, Amazon Bedrock            | Anthropic**, Amazon**                         |
+| Titan      | No            | Amazon Bedrock                       | Amazon\*\*                                    |
+
+_\*\* Connectors not yet available_
+
+Connectors may be needed not per Model basis but rather per platform.
+For example, using OpenAI or HuggingFace connector you may be able to call a Phi-2 Model.
+
+### Potential platforms for connector support:
+
+- **Local Model Support**
+  - LM Studio
+  - Ollama
+  - Pure C# (e.g. LlamaSharp or ONNX runtime)
+- Amazon Bedrock
+- Google AI Platform
+
+### Notes
+
+- Is very important to have easy to reproduce examples for any new model support or added feature.
+- While implementing new features code follow our current existing code-base as a reference and ensure that the code is validated by our CI/CD pipeline requirements, before submitting a PR for review.
+
+## Decision Outcome
+
+Chosen option: "Feature Branch Strategy", because it allows individual features to be developed in isolation, minimizing conflicts with the main branch and facilitating easier code reviews.
```

Comment:
We should also document our decisions on which connectors we plan to build and which deployment options we aim to support.